packages/rich-text/src/plugins/Hyperlink/HyperlinkModal.tsx

```diff
@@ -234,16 +228,29 @@ interface HyperLinkDialogData {
 
 export async function addOrEditLink(
   editor: ReactEditor & HistoryEditor & SPEditor,
-  { linkText, linkTarget, linkType, sdk }: addOrEditLinkProps
+  sdk: FieldExtensionSDK
 ) {
   if (!editor.selection) return;
 
+  let linkType;
+  let linkText;
+  let linkTarget;
+  let linkEntity;
+
+  const [node, path] = getNodeEntryFromSelection(editor, LINK_TYPES);
+  if (node) {
+    linkType = node.type;
+    linkText = (node.children[0] as TextElement).text;
```

Comment:
I guess we can use `Editor.string(editor, path)` to extract the text from the selection, I think it's safer.